### DISCUSSION TITLE
dzi memory usage

### DISCUSSION THREAD

**Discussion by tjdcs** [libvips/libvips#2098] [Q&A]
Hi all, 

Firstoff VIPS is an amazing image tool, easily one of my favorite projects. 

I have some very large images that are rendered as tiles at a single resolution  (so let's say like a 23x23 grid of 5k * 5k pngs) I want to stitch these together with vips and output to a dzi file. Last time I was working on this there was huge memory requirements for this workflow and I ended up renting a cloud machine with a huge stack of RAM for it. Unfortunately this is not always an option. 

Is there a better pipeline I can use or any chance at improving the memory requirements of this? I'm not immediately working on this project but will hopefully be rendering images next week. I just wanted to ask a little ahead of time since this was a problem last time.

```sh
#! /bin/bash
 cd  'base_tiles_folder'
name="bigImage_name"
rm -rf dzi/$name*
vips --vips-progress arrayjoin "$(ls final/*.tiff)" dzi/$name.dz[background=0,tile-size=720,suffix=.jpg[Q=80]] --across 23
 ```

**Comment by tjdcs** [comment]
Also (just a side note if there are interested readers) it seems like the deem zoom images is kind of a latent technology, vips and open sea dragon seem to be the most modern tools in the ecosystem but I'm not sure how much attention vips DZI gets or open sea dragon. Is this a dead technology / are there resources users that are still interested in deep zoom stuff?

**Comment by jcupitt** [comment]
Hi again @TuckerDowns,

Current libvips is a bit better at this, but I've not tested it recently. You'd need to experiment.

DZ still gets used a lot, but I think probably fewer people are doing it themselves. Thins like easyzoom have commoditized it for most users.

**Comment by tjdcs** [comment]
I will do further testing and report back.
